src/lib/BindToStores.react.js

```diff
@@ -1,47 +1,121 @@
-      )
-      actionClass[bootAction]()
+      // Create a listener for each resource store:
+      storeNames().forEach(store => {
+        const onChangeHandlerName = `handle${store}Change`
+        this[onChangeHandlerName] = this.onChangeFactory(store)
+        stores[store].addChangeListener(this[onChangeHandlerName])
+      })
+
+      // Fire the bootAction to fetch each resource:
+      var loadingStates = {}
+      bootObjects.forEach(bootObject => {
+        loadingStates[bootObject.loadingName] = true
+
+        /* eslint-disable indent */
+        // Choose which ID--if any--is passed to the GET request to identify the resource:
+        switch (bootObject.resourceType) {
+          case 'item':  // One specific item that MUST have an ID included in the GET request.
+            var idParam = this.props.params[bootObject.resourceId]
+            if (!idParam) { // Skip bootAction for items requiring ID if no ID URL param is available.
+              loadingStates[bootObject.loadingName] = false
+              break
+            }
+
+            bootObject.actionClass[bootObject.bootAction](idParam)
+            break
+          case 'itemAllParams':  // One specific item with a more-complex API endpoint URL as constructed in the action.
+            bootObject.actionClass[bootObject.bootAction](this.props.params)
+            break
+          case 'itemNoId':  // Single item that requires no ID (e.g., global config). Separate from list only to make it clear that it is only one item.
+          case 'list':  // List of items accessible at a single endpoint that does not require any resource ID.
+            bootObject.actionClass[bootObject.bootAction]()
+            break
+          default:  // Do not allow any resources of any other type to be requested--mainly a dev sanity check.
+            invariant(bootObject.resourceType, '`bootObject.resourceType` is unknown or undefined.')
+        }
+        /* eslint-enable indent */
+      }, this)
+      this.setState(loadingStates)
     },
-    componentWillUnmount() {
-      extractStores().forEach(store =>
-        store.removeChangeListener(this.handleStoresChanged)
-      )
+    componentWillUnmount() {  // Kill all listeners when component is unmounting.
+      storeNames().forEach(store => stores[store].removeChangeListener(this[`handle${store}Change`]))
     },
-    handleStoresChanged() {
-      var state = {processing: false, loading: false}
-      Object.getOwnPropertyNames(stores).map(store => {
-        state[store] = stores[store].getState()
-      })
-      this.setState(state)
+    goToThere(targetObject) {  // Push new router path if requested to do so in onBoundUpdate.
+      this.context.router.push(targetObject)
     },
-    render() {
-      return <Component {...this.props} {...this.state} />
-    }
+
+    onChangeFactory(store) {  // Create a custom listener identifier so loading states can be tracked independently for each resource.
+      var myStore = store
```

Comment:
I don't remember... though this smells of scope problems... perhaps a relic of some debugging and it's no longer neccessary. Anyway, let's clean it up once tests are in place.